src/main/java/mekanism/common/config/options/EnumOption.java

```diff
@@ -0,0 +1,107 @@
+package mekanism.common.config.options;
+
+import io.netty.buffer.ByteBuf;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import mekanism.common.config.BaseConfig;
+import net.minecraftforge.common.config.Configuration;
+import net.minecraftforge.common.config.Property;
+
+/**
+ * Created by Thiakil on 15/03/2019.
+ */
+@ParametersAreNonnullByDefault
+public class EnumOption<T extends Enum<T>> extends Option<EnumOption<T>> {
+
+    private T value;
+    private final T defaultValue;
+    private final Class<T> enumClass;
+    private final T[] enumValues;
+
+    @SuppressWarnings("unchecked")
+    public EnumOption(BaseConfig owner, String category, String key, T defaultValue, @Nullable String comment) {
+        super(owner, category, key, comment);
+        this.defaultValue = Objects.requireNonNull(defaultValue);
+        this.value = defaultValue;
+        this.enumClass = (Class<T>) defaultValue.getClass();
+        this.enumValues = enumClass.getEnumConstants();
+    }
+
+    public EnumOption(BaseConfig owner, String category, String key, T defaultValue) {
+        this(owner, category, key, defaultValue, null);
+    }
+
+    public T val() {
+        return value;
+    }
+
+    public void set(T value) {
+        this.value = Objects.requireNonNull(value);
+    }
+
+    @Override
+    public void load(Configuration config) {
+        final Property prop = config.get(this.category, this.key, this.defaultValue.name(), this.comment);
+
+        prop.setRequiresMcRestart(this.requiresGameRestart);
+        prop.setRequiresWorldRestart(this.requiresWorldRestart);
+
+        this.value = Enum.valueOf(this.enumClass, prop.getString());
+
+        //TODO: Special loading for enums, such as to default to RF and also allow aliases
```

Comment:
imo the aliases aren't necessary, only the enum names are ever saved so idk how someone would have the other names in the config. Default changing is just changing the value in the constructor call

Reply:
It isn't quite that simple because how we changed it to try and make it so you have to fully specify "joules". Also the aliases is in terms of the temp ones more.

Reply:
Though I am fine with just changing the default value for new configs if that is preferable.

Reply:
> It isn't quite that simple because how we changed it to try and make it so you have to fully specify "joules".

Could you elaborate on this?

Reply:
I think I will just change the default, except what I had meant is we had changed it so that "joules" has to be specified in the config otherwise it defaults to displaying "RF" as that is the number people are more familiar with what it means. Though I think it is fine until we switch internally to using FE that it shows Joules for the old installs.

Reply:
Yeah since it's display, probably best to not change existing installs.

As for changing to FE internally, that's probably something we need to discuss since some parts might need the Joules based storage